### PR TITLE
[labeler] Add engine workflow

### PR DESCRIPTION
**Pull request by CaseyHillers**
https://github.com/flutter/flutter/issues/126002

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

### CODE REVIEW

**Review by keyonghan — COMMENTED** [review]
Let's wait a bit until everything looks right in framework.

/cc @zanderso

File: .github/labeler.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
```
Comment (zanderso) [inline .github/labeler.yml:1]: You can also add the `impeller` label to changes under https://github.com/flutter/engine/tree/main/impeller

**Comment by CaseyHillers** [comment]
> Let's wait a bit until everything looks right in framework.

Happy to wait. To add context for Zach, landing this in the engine is much safer than the framework. The framework has a lot of custom Dart code for handling labels, so it's been hard to migrate everything over. The engine side was `path.startsWith(possiblePaths)` which gives me higher confidence in this landing.

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**